### ISSUE TITLE
GGMAX - Enhancement - Add emissive map and animated texture support to weapons

### ISSUE THREAD

**Issue by Kraven3D** [TheGameCreators/GameGuruRepo#5793]
For futuristic, sci-weapons with glowing elements such as light sabers and energy swords, to spinning, moving, glowing mechanical parts (animated textures should also support PBR):
 
![emissive_weapons](https://github.com/user-attachments/assets/98e68b23-87be-40c9-8fff-fb7822908adc)

**Comment by mav3r1ck1981** [comment]
i second this, and not only scifi weapons. modern day guns have holo sights, magical weapons and pickup definitely will need emissive support (wands/staffs/arrows/spells/hands etc) Long term i would assume that animated weapons will be implemented or at least i hope so! But an additional material slot for emissives on huds has been asked for alot over the years. Didnt classic have it?

**Comment by C4ever08** [comment]
I third this defo. This is something I've wanted for quite some time. I mean if particles can be added to weapons this should also be implemented.

**Comment by DBird-1994** [comment]
Emissive already works on weapons.  In fact I demonstrated this on discord like a year ago before one of your mods banned me.  Of course no one paid any attention because I wasn't one of the favored ones on discord.   This weapon is from sketchfab and using the machete animations from the wasteland dlc.  The blue parts use and emissive texture.

https://github.com/user-attachments/assets/239ef119-1453-4883-a37f-4392271273f0

**Comment by mav3r1ck1981** [comment]
@Kraven3D will have to see if Preben can add a slot to his weaponmaker for emissive textures if thats the case. Will do some testing later

**Comment by DBird-1994** [comment]
@mav3r1ck1981 Yes this is not a MAX issue since emissive already works, and I didn't use Preben's weapon importer when I made this a year ago.  So really this is more of an issue with Preben's tool if it's not working for you.

**Comment by mav3r1ck198-1** [comment]
It's not an issue per se, it only has 3 slots for _color _normal and _surface. There's nothing wrong with the tool. I'll do some testing to see if I can add them in via the fpe.

**Comment by Kraven3D** [comment]
There is no information regarding applying emissive maps to weapons in the official user guide or the weapons guide, and no official MAX weapons currently use them for me to reference, either, as far as I can tell. I would like _**official**_ emissive support for weapons to be added, not just jerry-rigging them to "sort-of" pull off the effect.

How are they set up? Can their emissive strength be adjusted so we have control over how bright they are? Do they cast light onto nearby objects? Are they reactive to bloom? Do we currently have any control over them via commands that can be added to the gunspec? Where are these commands I can reference? If not, I am requesting for such additions to be added. To take it one step further, adding animated texture support would give us even more options, such as flashing lights, pulsating lights and so on. Useful to make parts of the texture move whether or not the mesh they're applied to is animated or not. 

If all of this is already possible, great! I would absolutely love for it to be documented somewhere so I can reference it and start making use of it immediately. :) However, I suspect what we currently have to work with is limited to tossing an emissive map into the gamecore folder and hoping for the best, otherwise I could only assume more functionality would have been properly documented.

**Comment by C4ever08** [comment]
@Kraven3D Yes I agree this needs to be set up in Max core. And their is no real option to tweak the emission unless via FPE. It would be nice to have a slider. Or Preben adds a slot for this then that will be better.

**Comment by DBird-1994** [comment]
How are they set up?  -through the importer like every other model, or assign through your modeling program
Can their emissive strength be adjusted so we have control over how bright they are?  -yes, see above answer
Do they cast light onto nearby objects? - no
Are they reactive to bloom? - yes
Do we currently have any control over them via commands that can be added to the gunspec? -no

**Comment by Kraven3D** [comment]
@synchromesh62, thanks for this. Going to start making use of it immediately. These sort of things really should be added to the guide, though I understand it's probably hard for it to be kept up with since MAX keeps getting updated with new features. I'm sure there are other things that are possible in MAX already that I don't even know about for this same reason.

_(Going to leave this request open because animated texture support would still be very beneficial for use on weapons.)_

**Comment by synchromesh62** [comment]
> @synchromesh62, thanks for this. Going to start making use of it immediately. These sort of things really should be added to the guide, though I understand it's probably hard for it to be kept up with since MAX keeps getting updated with new features. I'm sure there are other things that are possible in MAX already that I don't even know about for this same reason.
> 
> _(Going to leave this request open because animated texture support would still be very beneficial for use on weapons.)_

As a side note for the weapon maker i used this process just to clarify how i got all the settings  
I imported the weapon first and setup the emissive on Import like any other model.
I then used that .dbo as the model in Melee Weapon Maker 
I then edited the weapons .fpe copying the emissive settings from the imported .fpe .. and added the emissive material to the gun folder.
Hope that helps.

**Comment by plemsoft** [comment]
Added emissive support to gunspec to control Player weapon emissive and NPC emissive. Pickup object can be controlled in normal way. Well sent it to @LeeBamberTGC tomorrow for possible inclusion in DevTest for a start :)

**Comment by synchromesh62** [comment]
> Added emissive support to gunspec to control Player weapon emissive and NPC emissive. Pickup object can be controlled in normal way. Well sent it to @LeeBamberTGC tomorrow for possible inclusion in DevTest for a start :) https://youtu.be/cHNjGFzPMwU

This is amazing !!
Thanks Preben
